File: classifier/data/raw/human/samples/project_euler_zig/problem_012.zig
// Project Euler 012: first triangular number with over 500 divisors.
const std = @import("std");

fn print(answer: u64) !void {
    var buffer: [32]u8 = undefined;
    const text = try std.fmt.bufPrint(&buffer, "{d}\n", .{answer});
    _ = std.c.write(1, text.ptr, text.len);
}

fn divisorCount(n: u64) u64 {
    var remaining = n;
    var factor: u64 = 2;
    var count: u64 = 1;
    while (factor * factor <= remaining) : (factor += 1) {
        var exponent: u64 = 0;
        while (remaining % factor == 0) {
            remaining /= factor;
            exponent += 1;
        }
        if (exponent != 0) count *= exponent + 1;
    }
    if (remaining > 1) count *= 2;
    return count;
}

pub fn main() !void {
    var index: u64 = 1;
    var triangle: u64 = 1;
    while (divisorCount(triangle) <= 500) {
        index += 1;
        triangle += index;
    }
    try print(triangle);
}

File: classifier/data/raw/ai/samples/project_euler_java/Problem012.java
// Project Euler Problem 12: first triangular number with over 500 divisors.
public class Problem012 {
    public static void main(String[] args) {
        for (long n = 1; ; n++) {
            long triangle = n * (n + 1) / 2;
            if (divisorCount(triangle) > 500) {
                System.out.println(triangle);
                return;
            }
        }
    }

    private static int divisorCount(long number) {
        int count = 1;
        for (long factor = 2; factor * factor <= number; factor += factor == 2 ? 1 : 2) {
            int exponent = 0;
            while (number % factor == 0) {
                number /= factor;
                exponent++;
            }
            count *= exponent + 1;
        }
        return count * (number > 1 ? 2 : 1);
    }
}
